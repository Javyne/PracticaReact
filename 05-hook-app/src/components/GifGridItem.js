import React from 'react'
import propType from 'prop-types'

export const GifGridItem = ({title, url}) => {
  
  
    return (
        <div className="card animate__animated animate__pulse">
            <img src={url} alt={title}/>
        </div>
  )
}

GifGridItem.propType = {
    title: propType.string.isRequired,
    url: propType.string.isRequired
}
