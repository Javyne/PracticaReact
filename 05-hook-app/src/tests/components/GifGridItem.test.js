import React from 'react';
import { shallow } from "enzyme"
import {GifGridItem} from "../../components/GifGridItem"

describe('test GifGridItem', () =>{
 
    const gif = {
        title: "titulo",
        url: "https://localhost/img.jpg"
    }
    const wrapper = shallow(<GifGridItem {...gif}/>)
    
    test('debe mosatar componente correctamente', () =>{
        
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener una imagen igual a la url y alt de los props', () =>{
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(gif.url)
        expect(img.prop('alt')).toBe(gif.title)
    })

    test('debe tener clase animate__pulse', () =>{
        
        const div = wrapper.find('div');

        expect(div.prop('className')).toContain('animate__pulse')

    })
})