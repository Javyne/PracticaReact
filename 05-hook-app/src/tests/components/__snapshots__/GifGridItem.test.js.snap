// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifGridItem debe mosatar componente correctamente 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="titulo"
    src="https://localhost/img.jpg"
  />
</div>
`;
